src/main/java/spark/route/RouteEntry.java

```diff
@@ -95,33 +97,28 @@ private boolean matchPath(String path) { // NOSONAR
         } else {
             // Number of "path parts" not the same
             // check wild card:
-            if (this.path.endsWith("*")) {
-                if (pathSize == (thisPathSize - 1) && (path.endsWith("/"))) {
-                    // Hack for making wildcards work with trailing slash
-                    pathList.add("");
-                    pathList.add("");
-                    pathSize += 2;
-                }
+            if (this.path.contains("*") && thisPathSize < pathSize) {
+                StringBuilder sb = new StringBuilder();
+                char[] charArray = this.path.toCharArray();
+                for (int i = 0 ; i < charArray.length ; ++i) {
+                    if (charArray[i] == '*') {
+                        sb.append('.');
+                    }
 
-                if (thisPathSize < pathSize) {
-                    for (int i = 0; i < thisPathSize; i++) {
-                        String thisPathPart = thisPathList.get(i);
-                        String pathPart = pathList.get(i);
-                        if (thisPathPart.equals("*") && (i == thisPathSize - 1) && this.path.endsWith("*")) {
-                            // wildcard match
-                            return true;
-                        }
-                        if (!thisPathPart.startsWith(":")
-                                && !thisPathPart.equals(pathPart)
-                                && !thisPathPart.equals("*")) {
-                            return false;
+                    if (charArray[i] == ':') {
+                        while (charArray[++i] != '/') {
                         }
+                        sb.append(".*");
                     }
-                    // All parts matched
-                    return true;
+
+                    sb.append(charArray[i]);
                 }
-                // End check wild card
+                String thisPath = sb.toString();
+                Pattern pattern = Pattern.compile(thisPath);
+                Matcher matcher = pattern.matcher(path);
+                return matcher.matches();
```

Comment:
Hey @haspriyank I would add tests cases with sufficiently large urls just to see how it behaves. Java matcher can be quite slow specially with this type of regex....

Reply:
More 1 point is considering caching the compiled pattern, since the matchers will be evaluated at every request, just after changing the url on spark notation (using `*` as wildcard) to java notation (using `.*`)...

Reply:
An interesting article: https://www.javaworld.com/article/2077757/core-java/optimizing-regular-expressions-in-java.html